### ISSUE TITLE
ds-ctcdecoder not working well in Chinese characters (v0.6.0)

### ISSUE THREAD

**Issue by iekhwang** [mozilla/DeepSpeech#2860]
For support and discussions, please use our [Discourse forums](https://discourse.mozilla.org/c/deep-speech).

If you've found a bug, or have a feature request, then please create an issue with the following information:

- **Have I written custom code (as opposed to running examples on an unmodified clone of the repository)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **TensorFlow installed from (our builds, or upstream TensorFlow)**: official Dockerfile
- **TensorFlow version (use command below)**:1.14.0
- **Python version**: 3.6.9
- **Bazel version (if compiling from source)**: Build label: 0.24.1
- **GCC/Compiler version (if compiling from source)**: 7.5.0
- **CUDA/cuDNN version**: 10.0
- **GPU model and memory**: Titan xp 
- **Exact command to reproduce**:

Last year I have trained a Mandarin model with v0.6.0-alpha4. When I try to use stable version v0.6.0, the model only can output one Chinese character in the decoder phase. 
![WhatsApp Image 2020-03-28 at 19 16 12](https://user-images.githubusercontent.com/39388603/77833909-0d68b400-7141-11ea-9bdb-3d1d0c5ad59a.jpeg)

After that, I have tested one corpus with v0.6.0alpha4 - v0.6.0alpha15 by using corresponds ds-ctcdecoder. The result of v0.6.0alpha4 - v0.6.0alpha12 are the same like last year I trained, but after v0.6.0alpha13,  the CER gets worse. 
<img width="962" alt="Screenshot 2020-03-28 at 22 18 49" src="https://user-images.githubusercontent.com/39388603/77834127-8288b900-7142-11ea-9cb5-4b8061ec422c.png">
<img width="962" alt="Screenshot 2020-03-28 at 22 22 46" src="https://user-images.githubusercontent.com/39388603/77834178-f1fea880-7142-11ea-8181-153075ce1d30.png">

In v0.6.0alpha15 and the stable version v0.6.0, the model just can output one character in each sentence likes my first picture. 
The experiments are in docker container generated by Dockerfile in master. All parameters are the same in my tests(lm_alpha and lm_beta are default setting). I noticed the losses are the same. I think it's not the problem in the training part, so I am wondering if there are some big changes in ctcdecoder after v0.6.0alpha13 that not friendly for character-based language training?

**Comment by reuben** [comment]
Alpha versions have no guarantee of stability between them. I don't know what happened between those alphas but in general if your acoustic model was trained on a specific version you should stick to that version on all components, not mix and match versions of the model and the client or the decoder. Basically, until 1.0 is out, stick to the same version of every component: version of training/export code, decoder package, version of native client, version of acoustic model/checkpoint if using our releases, version of any tooling like `generate_trie`, etc.

**Comment by reuben** [comment]
FYI, there is one big change that affects character based models, but it only landed in v0.6.0-alpha.15: UTF-8 targeting, which supersedes the old character based decoding paths.

**Comment by iekhwang** [comment]
> Alpha versions have no guarantee of stability between them. I don't know what happened between those alphas but in general if your acoustic model was trained on a specific version you should stick to that version on all components, not mix and match versions of the model and the client or the decoder. Basically, until 1.0 is out, stick to the same version of every component: version of training/export code, decoder package, version of native client, version of acoustic model/checkpoint if using our releases, version of any tooling like `generate_trie`, etc.

Thanks for your quick reply. Yes, I am sure that when I used the stable version v0.6.0, I have cloned the original code, and used the Dokerfile in masker generated the docker image and docker container. Native client, ctcdecoder, and generated_trie are all based on the source code. The result has encountered the problem which I showed in the first picture(only one character in each sentence). As you mentioned, it may because of the UTF-8 targeting. I also noticed the `_swigwrapper.cpython-36m-x86_64-linux-gnu.so` file is larger than the old version,  after alpha13, a new Alphabet class is introduced as well.
